pkg/metrics/osbclientproxy/osbproxy.go

```diff
@@ -132,21 +142,48 @@ func (pc proxyclient) Unbind(r *osb.UnbindRequest) (*osb.UnbindResponse, error)
 func (pc proxyclient) GetBinding(r *osb.GetBindingRequest) (*osb.GetBindingResponse, error) {
 	glog.V(9).Info("OSBClientProxy GetBinding()")
 	response, err := pc.realOSBClient.GetBinding(r)
-	pc.updateMetrics(err)
+	pc.updateMetrics(getBinding, err)
 	return response, err
 }
 
-// updateMetrics bumps the request count metric for the specific broker and
-// status
-func (pc proxyclient) updateMetrics(err error) {
+const (
+	oneXX     = "1xx"
+	twoXX     = "2xx"
+	threeXX   = "3xx"
+	fourXX    = "4xx"
+	fiveXX    = "5xx"
+	clientErr = "client-error"
+)
+
+// updateMetrics bumps the request count metric for the specific broker, method
+// and status
+func (pc proxyclient) updateMetrics(method string, err error) {
+	var statusGroup string
+
 	if err == nil {
-		metrics.OSBRequestCount.WithLabelValues(pc.brokerName, "200").Inc()
-	} else {
-		status, ok := osb.IsHTTPError(err)
-		if ok {
-			metrics.OSBRequestCount.WithLabelValues(pc.brokerName, strconv.Itoa(status.StatusCode/100*100)).Inc()
-		} else {
-			metrics.OSBRequestCount.WithLabelValues(pc.brokerName, "client-error").Inc()
+		metrics.OSBRequestCount.WithLabelValues(pc.brokerName, method, twoXX).Inc()
+		return
+	}
+
+	status, ok := osb.IsHTTPError(err)
+	if ok {
+		switch status.StatusCode / 100 {
```

Comment:
This seems a little convoluted to me. Why not just do `fmt.Sprintf("%dxx", status.StatusCode/100)`?

Reply:
Coming from a java world, performance wise this would result in the least string operations.  Maybe that isn't such a big issue in Go?  I agree the Sprintf is a lot cleaner.

Reply:
Even in Java, it would take a lot of convincing to persuade me that the minimal performance implications of Sprintf are worth the cost of code cleanliness in this situation. This code is called right before building up an HTTP request, including JSON marshalling in many cases; making a blocking HTTP call; and doing JSON un-marshalling on the response. A Sprintf here is of very small consequence in comparison.

Reply:
I agree Matt, thanks for the review and comments, corrections will be posted momentarily.